cirq-core/cirq/ops/eigen_gate_test.py

```diff
@@ -363,6 +363,12 @@ def _with_exponent(self, exponent):
         (cirq.X, cirq.Y, False),
         (cirq.rz(np.pi), cirq.Z, True),
         (cirq.X**0.3, cirq.Z**0.3, False),
+        (cirq.Z, cirq.Z ** (1 - 1e-10), True),
+        (cirq.Z, cirq.Z ** (1 + 1e-10), True),
+        (cirq.Z**2, cirq.Z ** (2 - 1e-10), True),
+        (cirq.Z**2, cirq.Z ** (2 + 1e-10), True),
+        (cirq.Z, cirq.Z ** (3 - 1e-10), True),
```

Comment:
nit: Does this test for any behavior different from the `1 - 1e10` case above?

Reply:
Yes, the critical part is to verify the exponents are recognized as equivalent w/r to a period if their difference is eps-larger or eps-smaller than the period.  This was not right in master and not quite right in the first iteration here.
Taking a second look, it would make more sense to compare (Z**2, Z**(4±eps)) so I am going to update it that way.